### DISCUSSION TITLE
Processed image from rawtherapee gets not stacked

### DISCUSSION THREAD

**Discussion by xsteadfastx** [photoprism/photoprism#3725] [Q&A] [answered]
i download my raw file from photoprism and processed it locally in rawtherapee. now i wanted to add this to the raw stack. i upload the new file via import. create date in exit tool are the same and the filename base is also the same:

```
❯ exiftool 20230910_140148_8D91F63A.jpg|grep Date
File Modification Date/Time     : 2023:09:14 11:47:18+00:00
File Access Date/Time           : 2023:09:14 11:58:23+00:00
File Inode Change Date/Time     : 2023:09:14 11:58:23+00:00
Modify Date                     : 2023:09:10 14:01:48
Date/Time Original              : 2023:09:10 14:01:48
Create Date                     : 2023:09:10 14:01:48
Profile Date Time               : 2018:11:13 11:18:51

❯ exiftool 20230910_140148_8D91F63A.arw|grep Date
File Modification Date/Time     : 2023:09:14 11:44:27+00:00
File Access Date/Time           : 2023:09:14 11:45:27+00:00
File Inode Change Date/Time     : 2023:09:14 11:44:28+00:00
Date/Time Original              : 2023:09:10 14:01:48
Create Date                     : 2023:09:10 14:01:48
Sony Date Time                  : 2023:09:10 14:01:48
Sony Date Time 2                : 2023:09:10 12:01:48
Modify Date                     : 2023:09:10 14:01:48
```

how can i stack them? or what would be the best workflow to work with photoprism and local photo editing? do i have to add this file to the original folder because the import will create a new name for it?

here is a screenshot of the edited and uploaded image:

![image](https://github.com/photoprism/photoprism/assets/168397/3bc0dc68-a0c1-4b5c-874a-b015ee8d752f)

new name but original name is there.

and the original here:

![image](https://github.com/photoprism/photoprism/assets/168397/7b03c3ce-ae56-4d69-9148-b629a02d5086)

**Comment by graciousgrey** [comment, marked as answer]
> do i have to add this file to the original folder because the import will create a new name for it?

Exactly, when you import the files they get a new name, so the files in the originals folder have different names and are not stacked.

Instead of using the web upload, you could add the file (20230910_140148_8D91F63A.jpg) next to 20230910_140148_8D91F63A.arw in the original folder and trigger indexing for that folder.

**Reply by xsteadfastx** [reply]
ah ok. now im trying just to open the file through webdav originals and store it next to it like you wrote. the only problem for me is to name it correctly that it gets stacked up.  i wish photoprism would also accept something like `20230910_140148_8D91F63A.bw.jpg` or `20230910_140148_8D91F63A.arw.bw.jpg`.

**Reply by graciousgrey** [reply]
For now you could try `20230910_140148_8D91F63A_bw.jpg `

**Reply by xsteadfastx** [reply]
this doesnt work. this creates a new entry and doesnt stack to its original. just tested it.

**Reply by graciousgrey** [reply]
Stacking by sequential name needs to be enabled for this: https://docs.photoprism.app/user-guide/settings/library/#stacks. Please note that this setting will apply to your whole library so only use it, if you don't expect to get a lot of files stacked you don't want to have stacked.

**Reply by xsteadfastx** [reply]
i have it enabled:

![image](https://github.com/photoprism/photoprism/assets/168397/0d8e731d-0409-4248-bc76-cfa929ba34aa)

i added the jpg here:

![image](https://github.com/photoprism/photoprism/assets/168397/04079d5b-c187-4764-b36d-66a2ffded460)

but it doesnt stack it to its original:

![image](https://github.com/photoprism/photoprism/assets/168397/ecb050de-1915-4a8e-9f68-48bdeb929eef)

its just the auto generated jpg from photoprism:

![image](https://github.com/photoprism/photoprism/assets/168397/e3464cf7-38b7-4e3d-9a1c-d1f1b2a74354)

**Reply by lastzero** [reply]
It's because "out" isn't ignored when comparing the filenames, as it is not a supported / common file extension. Note that some users have dots in their filenames as part of the actual name, for example `Wow... what a crazy day!.jpg`.

**Reply by xsteadfastx** [reply]
so also the `_bw.jpg` gets ignored? sorry i really try to get my head around it. the pp3 file come from rawtherapee and just adds the `out` to the pp3 filename. so the `_bw.jpg` gets ignored because there is a pp3 to it that has an `out` in it?

**Reply by lastzero** [reply]
Oh no, sorry I just had a quick look on my phone... there's a couple of patterns that get ignored, like `.1.` and ` (2)`, but not `_bw`. So that might be another issue then!

**Reply by lastzero** [reply]
The list with filename examples at https://docs.photoprism.app/user-guide/organize/stacks/#for-what-reasons-can-files-be-stacked unfortunately isn't very extensive — see the following test suite for what patterns work (and have been tested): 
- https://github.com/photoprism/photoprism/blob/develop/pkg/fs/base_test.go

**Reply by graciousgrey** [reply]
I will update the user guide as soon as possible :+1: 

It indeed seems that _bw does not work in all cases.

In my tests, `20230910_140148_8D91F63A_bw.jpg` was stacked with `20230910_140148_8D91F63A.jpg` and `20230910_140148_8D91F63A.arw` if all files were present at initial indexing.

If `20230910_140148_8D91F63A.arw` was previously indexed so that PhotoPrism created a `20230910_140148_8D91F63A.arw.jpg` file, `20230910_140148_8D91F63A_bw.jpg` will not be stacked if added later.

**Reply by lastzero** [reply]
Yep, might depend on which file is the primary, since the base name of the main file is compared. Related sidecar files may have longer names than the main file, but again it depends on the pattern.

**Reply by xsteadfastx** [reply]
its such a good idea just to look at the code. and now i understand it a little bit better. thank you :)
